File: Raj Bhushan/Angular/src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Router } from '@angular/router';
import { Customer } from '../entity/Customer';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http:HttpClient , private router:Router) { }

  url:string="http://localhost:3300/ebms"

  //Code for viewing the bill generated on customer
  viewBillHistory(customerId:number):Observable<any>
  {
     return this.http.get(this.url+'/viewBillHistory/'+customerId)
  }
  
  //Code for viewing the payment status of customer
  viewPaymentHistory(customerId:number):Observable<any>
  {
    return this.http.get(this.url+'/viewPaymentHistory/'+customerId)
  }

  //Code for viewing the profile
  viewCustomerProfile(customerId:number):Observable<any>
  {
    return this.http.get(this.url+'/viewCustomerProfile/'+customerId)
  }
  
  //Code for updating the profile 
  editProfile(customerId:number,customer:Customer):Observable<any>
{
    return this.http.put(this.url+'/editProfile/'+customerId+"/", customer);
}

 //Code for changing the password
 changePassword(customerId:number,oldPassword:string, newPassword:string):Observable<any>
 {
   return this.http.get(this.url+'/changePassword/'+customerId+"/"+oldPassword+"/"+newPassword)
 }
}

File: Raj Bhushan/Springboot/src/main/java/com/capgemini/ebms/service/CustomerService.java
package com.capgemini.ebms.service;

import com.capgemini.ebms.beans.CustomerDetail;
import com.capgemini.ebms.beans.RegistrationModel;
import com.capgemini.ebms.exception.UserException;

public interface CustomerService {
    //For registering the customer
	CustomerDetail register(RegistrationModel registerDetails) throws UserException;
    
	//For login the customer
	Long login(long custId, String custPassword) throws UserException;
    
	//For viewing the profile of customer
	CustomerDetail viewCustomerProfile(long custId) throws UserException;
    
	//For editing the profile of customer
	CustomerDetail editProfile(long custId, CustomerDetail customer) throws UserException;
    
	//For changing the password
	boolean changePassword(long custId, String oldPassword, String newPassword) throws UserException;

}
